Sources/macros.js

```diff
@@ -391,6 +391,12 @@ export function obj(publicAPI = {}, model = {}) {
     publicAPI.modified();
   };
 
+  // This function will get called when one invoke JSON.stringify(vtkObject)
+  // JSON.stringify will only stringify the return value of this function
+  publicAPI.toJSON = function vtkObjToJSON() {
```

Comment:
why not write `publicAPI.toJSON = publicAPI.getState`

Reply:
if `getState()` get overriden in subclass

Reply:
Make sense...